pkg/aws/aws.go

```diff
@@ -39,26 +39,37 @@ type EC2Client interface {
 	SubnetClient
 }
 
-// GetEC2Client returns and EC2Client along with the VPC id and the cluster name
-func GetEC2Client(ctx context.Context, client client.Client) (EC2Client, string, string, error) {
+// GetEC2Client returns and EC2Client along with the VPC id, cluster name and AWS region
+func GetEC2Client(ctx context.Context, client client.Client) (ec2Client EC2Client, vpcID string, clusterName string, awsRegion string, err error) {
```

Comment:
I think we can have additional methods `GetAWSRegion()`,  etc.  defined on the interface and implemented on struct to share required objects, slightly more decoupled. WDYT?

Reply:
So the implementing struct is not owned by us. It's actually the EC2 client returned by the SDK. Unfortunately there's no easy way to mock the client. What I have implemented is how they recommend the client be mocked. That said I've refactored the code so that all the information is not returned from a single function. I've split it into 3 functions and added tests for one of them.

Reply:
I actually meant in a new struct, like this 
```go
type AWS struct {
	infra       configv1.Infrastructure
	clustername string
}

func (a AWS) New() AWS {
 // init clustername, etc
}

func (a AWS) GetEC2Client() (EC2Client, error) {
	panic("")
}

func (a AWS) GetVPC() string {
	panic("")
}

func (a AWS) GetRegion() string {
	panic("")
}
```
But splitting it up works as well, I think. For tests I think we can use `gomock` and generate something like [this](https://github.com/openshift/aws-load-balancer-controller/tree/main/mocks/controller-runtime/client) (kind of an overkill for this though).